### PR TITLE
risc-v: Implement common up_cpu_idlestack

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
* risc-v: Implement common up_cpu_idlestack
* Add -mstrictalign to k210's arch level flags since it don't support unaligned memory access
* Use common cpu idle stack implementation for k210
## Impact
k210 smp only.
## Testing
K210

### CODE REVIEW

File: arch/risc-v/src/common/riscv_cpuidlestack.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+#include <nuttx/compiler.h>
+#include <nuttx/sched.h>
+
+#include "riscv_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define SMP_STACK_MASK       7
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_cpuidlestack.c:40]: 7->15

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@no1wudi 
My automatic test failed due to this PR.
Could you please try maix-bit:smp with QEMU?

```
NuttShell (NSH) NuttX-10.2.0
nsh>  
nsh> uname -a
NuttX 10.2.0 b8477f857b Feb 14 2022 16:09:56 risc-v maix-bit
nsh> mount
  /proc type procfs
nsh> ps
  PID GROUP CPU PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK   STACK   USED  FILLED COMMAND
    0     0   0   0 FIFO     Kthread N-- Assigned           00000000 002000 000840  42.0%  CPU0 IDLE
    1     1   1   0 FIFO     Kthread N-- Running            00000000 002000 002000 100.0%! CPU1 IDLE
    2     2   0 100 RR       Task    --- Running            00000000 003040 001824  60.0%  nsh_main
nsh> free
                   total       used       free    largest  nused  nfree
        Umem:    2082112      12112    2070000    2070000     35      1
nsh> hello
Hello, World!!
Traceback (most recent call last):
  File "/home/ishikawa/script/expect_nuttx_smp_test.py", line 38, in <module>
    child.expect('nsh> ')
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/spawnbase.py", line 344, in expect
    timeout, searchwindowsize, async_)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/spawnbase.py", line 372, in expect_list
    return exp.expect_loop(timeout)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/expect.py", line 181, in expect_loop
    return self.timeout(e)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/expect.py", line 144, in timeout
    raise exc
pexpect.exceptions.TIMEOUT: Timeout exceeded.
```

**Comment by no1wudi** [comment]
@masayuki2009 
I've it tested on real device , tried your operation but it seems ok and `ostest` and `smp` test also works fine.
```
NuttShell (NSH) NuttX-10.2.0
nsh> uname  -a
NuttX 10.2.0 b8477f857b Feb 14 2022 15:20:50 risc-v maix-bit
nsh> mount
  /proc type procfs
nsh> ps
  PID GROUP CPU PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK   STACK   USED  FILLED COMMAND
    0     0   0   0 FIFO     Kthread N-- Assigned           00000000 002000 000840  42.0%  CPU0 IDLE
    1     1   1   0 FIFO     Kthread N-- Running            00000000 002000 000048   2.4%  CPU1 IDLE
    2     2   0 100 RR       Task    --- Running            00000000 003040 002008  66.0%  nsh_main
nsh> free
                   total       used       free    largest  nused  nfree
        Umem:    2081632      12112    2069520    2069520     35      1
nsh> hello
Hello, World!!
nsh>
```

**Comment by no1wudi** [comment]
@masayuki2009 
I notice that the avaliable memory is different on your qemu and my real board, could you have check for it ?

**Comment by masayuki2009** [comment]
> @masayuki2009 I notice that the avaliable memory is different on your qemu and my real board, could you have check for it ?

@no1wudi 
You can use qemu-5.2.

```
$ qemu-system-riscv64 -nographic -machine sifive_u -bios nuttx

Without this PR
NuttShell (NSH) NuttX-10.2.0
nsh> ps
  PID GROUP CPU PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK   STACK   USED  FILLED COMMAND
    0     0   0   0 FIFO     Kthread N-- Assigned           00000000 002000 000840  42.0%  CPU0 IDLE
    1     1   1   0 FIFO     Kthread N-- Running            00000000 002000 000048   2.4%  CPU1 IDLE
    2     2   0 100 RR       Task    --- Running            00000000 003040 001808  59.4%  nsh_main
nsh> hello
Hello, World!!
nsh>
```

**Comment by masayuki2009** [comment]
@no1wudi 
As you can see, the stack for the CPU1 IDLE task is corrupted.

**Comment by no1wudi** [comment]
> @no1wudi As you can see, the stack for the CPU1 IDLE task is corrupted.

The cpux's idle stack is loaded from `g_cpu_basestack` (data section) now, but on this time it maybe not ready since it is initialized by cpu0, I'll provide another PR to fix it.

**Comment by no1wudi** [comment]
@masayuki2009 Please see #5490

File: arch/risc-v/src/common/riscv_cpuidlestack.c [hunk 2]
```diff
@@ -28,24 +28,86 @@
...
+
+#if CONFIG_SMP_NCPUS > 7
+static uint8_t aligned_data(16) cpu7_idlestack[CONFIG_IDLETHREAD_STACKSIZE];
+#endif
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+uint8_t *g_cpu_basestack[CONFIG_SMP_NCPUS] =
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_cpuidlestack.c:86]: the problem can be fixed it by add const instead.

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_cpuidlestack.c:86]: Sure， good idea

**Pull request merged**